src/expr.cc

```diff
@@ -371,12 +372,31 @@ MemberLookup::MemberLookup(Expression *expr, const std::string &member, const Lo
 Value MemberLookup::evaluate(const std::shared_ptr<const Context>& context) const
 {
     const Value &v = this->expr->evaluate(context);
+    static const boost::regex re_swizzle_validation("^([xyzw]{1,4}|[rgba]{1,4})$");
```

Comment:
I would use `{2,4}` here just to be precise, even though I realize it is inconsequential since `member.length() > 1` is also checked below.